packages/flutter_tools/bin/xcode_debug.js

```diff
@@ -0,0 +1,519 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/**
+ * @fileoverview OSA Script to interact with Xcode. Functionality includes
+ * checking if a given project is open in Xcode, starting a debug session for
+ * a given project, and stopping a debug session for a given project.
+ */
+
+'use strict';
+
+function run(args_array = []) {
```

Comment:
I'm guessing this gets interpreted by the invocation of osascript from xcode_debug.dart. Where does execution start? Does osascript invoke `run()`, forwarding it's CLI args?

Reply:
So `run` is a key function that gets run when the script is run (https://developer.apple.com/library/archive/documentation/LanguagesUtilities/Conceptual/MacAutomationScriptingGuide/UseHandlersFunctions.html#:~:text=certain%20events%20occur.-,Run%20Handlers,-The%20run%20event).

Also, `man osascript` states "Any arguments following the script will be passed as a list of strings to the direct parameter of the “run” handler."

Reply:
I can add a comment to the function to explain it